pkg/cli/adbtunnel.go

```diff
@@ -122,11 +132,68 @@ func openADBTunnel(flags *ADBTunnelFlags, c *command, args []string, opts *subCo
 
 func listADBTunnels(flags *listADBTunnelFlags, c *command, opts *subCommandOpts) error {
 	controlDir := opts.InitialConfig.ADBControlDirExpanded()
+	forwarders, err := gatherADBTunnels(controlDir, flags.Zone, flags.host)
+	for _, status := range forwarders {
+		line := fmt.Sprintf("%s/%s 127.0.0.1:%d %s",
+			status.Host, status.Device, status.Port, status.State)
+		if flags.longFormat {
+			line = fmt.Sprintf("%s: %s/%s", status.ServiceURL, status.Zone, line)
+		}
+		c.Println(line)
+	}
+	return err
+}
+
+func closeADBTunnels(flags *closeADBTunnelFlags, c *command, args []string, opts *subCommandOpts) error {
+	devices := make(map[string]any)
+	for _, a := range args {
+		devices[a] = nil
+	}
+	controlDir := opts.InitialConfig.ADBControlDirExpanded()
+	forwarders, merr := gatherADBTunnels(controlDir, flags.Zone, flags.host)
+	if len(args) > 0 {
+		var inArgs []ADBForwarderStatus
+		for _, fwdr := range forwarders {
+			if _, in := devices[fwdr.Device]; in {
+				inArgs = append(inArgs, fwdr)
+			}
+		}
+		forwarders = inArgs
+	}
+	if len(forwarders) == 0 {
+		return fmt.Errorf("No ADB tunnels found")
+	}
+	if len(forwarders) > 1 && !flags.skipConfirmation {
+		var err error
+		forwarders, err = askUserToChoose(forwarders, c)
+		if err != nil {
+			// A failure to read user input cancels the entire command.
+			return err
+		}
+	}
+	for _, dev := range forwarders {
+		conn, err := net.Dial("unixpacket", fmt.Sprintf("%s/%d.sock", controlDir, dev.Port))
```

Comment:
can we put this logic into a single function: `closeADBTunnel`?

Reply:
Done